modules/mobile/app/controllers/mobile/v0/payment_history_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require_dependency 'mobile/application_controller'
+require 'adapters/payment_shared_adapter'
+
+module Mobile
+  module V0
+    class PaymentHistoryController < ApplicationController
+      def index
+        start_date = params[:startDate] || (DateTime.now.utc.beginning_of_day - 1.year).iso8601
+        end_date = params[:endDate] || DateTime.now.utc.beginning_of_day.iso8601
+
+        validated_params = Mobile::V0::Contracts::GetPaginatedList.new.call(
+          start_date: start_date,
+          end_date: end_date,
+          page_number: params.dig(:page, :number),
+          page_size: params.dig(:page, :size),
+          use_cache: false,
+          reverse_sort: false
+        )
+
+        person = BGS::PeopleService.new(current_user).find_person_by_participant_id
+        response = BGS::PaymentService.new(current_user).payment_history(person)
+        payments = adapt_payments(response.dig(:payments, :payment))
+        list, meta = paginate(payments, validated_params)
+        render json: Mobile::V0::PaymentHistorySerializer.new(list, meta)
+      end
+
+      private
+
+      def adapt_payments(payments)
+        payments.map do |payment|
+          if payment.dig(:return_payment, :check_trace_number).blank?
```

Comment:
Having a conditional within a map like this will leave nils when the condition isn't met. You may need to change how you're doing this or use `compact` to get rid of the nils.

Reply:
Thanks! - made the correction